test/contracts-network/metatx-token.js

```diff
@@ -0,0 +1,725 @@
+/* globals artifacts */
+import chai from "chai";
+import bnChai from "bn-chai";
+import { ethers } from "ethers";
+import { web3GetBalance, checkErrorRevert, expectEvent } from "../../helpers/test-helper";
+import { getMetatransactionParameters, getPermitParameters } from "../../helpers/test-data-generator";
+
+const { expect } = chai;
+chai.use(bnChai(web3.utils.BN));
+
+const EtherRouter = artifacts.require("EtherRouter");
+const IColony = artifacts.require("IColony");
+const IColonyNetwork = artifacts.require("IColonyNetwork");
+const MetaTxToken = artifacts.require("MetaTxToken");
+
+const ADDRESS_ZERO = ethers.constants.AddressZero;
+
+contract("MetaTx token", (addresses) => {
+  let colony;
+  let metatxToken;
+  let colonyNetwork;
+
+  before(async () => {
+    const etherRouter = await EtherRouter.deployed();
+    colonyNetwork = await IColonyNetwork.at(etherRouter.address);
+  });
+
+  beforeEach(async () => {
+    metatxToken = await MetaTxToken.new("Test", "TEST", 18);
+    const { logs } = await colonyNetwork.createColony(metatxToken.address, 0, "", "");
+    const { colonyAddress } = logs.filter((x) => x.event === "ColonyAdded")[0].args;
+    colony = await IColony.at(colonyAddress);
+    await colony.setRewardInverse(100);
+  });
+  describe("when using the contract directly", () => {
+    describe("when working with MetaTxToken, should behave like normal token", () => {
+      beforeEach("mint 1500000 tokens", async () => {
+        await metatxToken.unlock({ from: addresses[0] });
```

Comment:
Can we add `USER0 = addresses[0]` at the start of the file? Also why not `accounts` like we do elsewhere?